src/jsonb_utils.c

```diff
@@ -57,6 +57,45 @@ ts_jsonb_add_str(JsonbParseState *state, const char *key, const char *value)
 	ts_jsonb_add_value(state, key, &json_value);
 }
 
+void
+ts_jsonb_set_value_by_type(JsonbValue *value, Oid typeid, Datum datum)
+{
+	switch (typeid)
+	{
+		Oid typeOut;
+		bool isvarlena;
+		char *str;
+
+		case INT8OID:
+			value->type = jbvNumeric;
+			value->val.numeric = DatumGetNumeric(DirectFunctionCall1(int8_numeric, datum));
+			break;
+
+		case INT2OID:
+			value->type = jbvNumeric;
+			value->val.numeric = DatumGetNumeric(DirectFunctionCall1(int2_numeric, datum));
+			break;
+
+		case INT4OID:
+			value->type = jbvNumeric;
+			value->val.numeric = DatumGetNumeric(DirectFunctionCall1(int4_numeric, datum));
+			break;
+
+		case NUMERICOID:
+			value->type = jbvNumeric;
+			value->val.numeric = DatumGetNumeric(datum);
+			break;
+
+		default:
```

Comment:
Is it possible to get test coverage for these cases? Many seem to be uncovered. However, most are just integer variants, but the `default` case looks completely different.

Reply:
I added test coverage by rewriting one of the existing function to use the new type-based function. PTAL.